torch/quantization/observer.py

```diff
@@ -689,26 +689,30 @@ def _load_from_state_dict(
         unexpected_keys: List[str],
         error_msgs: List[str],
     ):
-        local_state = ["min_vals", "max_vals"]
+        version = local_metadata.get("version", None)
+        if version is None or version < 3:
+            local_state = ["min_vals", "max_vals"]
+        else:
+            local_state = ["min_val", "max_val"]
         for name in local_state:
             key = prefix + name
             if key in state_dict:
                 val = state_dict[key]
-                # Custom handling to allow loading min_vals or max_vals
+                # Custom handling to allow loading min_val or max_val
                 # of size N into uninitialized buffers of size 0. The
                 # buffers are resized here, and the values are copied in
                 # the default state_dict loading code of the parent.
-                if name == "min_vals":
-                    self.min_vals.resize_(val.shape)
+                if name == "min_val" or name == "min_vals":
+                    self.min_val.resize_(val.shape)
                 else:
-                    self.max_vals.resize_(val.shape)
+                    self.max_val.resize_(val.shape)
                 # For torchscript module we need to update the attributes here since we do not
                 # call the `_load_from_state_dict` function defined module.py
                 if torch.jit.is_scripting():
-                    if name == "min_vals":
-                        self.min_vals.copy_(val)
+                    if name == "min_val" or name == "min_vals":
+                        self.min_val.copy_(val)
                     else:
-                        self.max_vals.copy_(val)
+                        self.max_val.copy_(val)
```

Comment:
does this implicitly handle the conversion?  Would it be better to do it explicitly, something like "if version < N, read from `min_vals` and move to `min_val`?  Otherwise it's hard to see what exactly is happening for the casual reader

Reply:
I tried making it a bit more easier to read while still trying to minimize code duplication